packages/core/src/main/cryptography/CryptographyService.ts

```diff
@@ -84,30 +74,10 @@ export class CryptographyService {
       .filter(serializedPreKey => serializedPreKey.key);
   }
 
-  public async decrypt(sessionId: string, encodedCiphertext: string): Promise<DecryptionResult> {
+  public async decrypt(sessionId: string, encodedCiphertext: string): Promise<Uint8Array> {
     this.logger.log(`Decrypting message for session ID "${sessionId}"`);
     const messageBytes: Uint8Array = Decoder.fromBase64(encodedCiphertext).asBytes;
-
-    try {
-      const result = await this.cryptobox.decrypt(sessionId, messageBytes.buffer);
-      return {
-        isSuccess: true,
-        value: result,
-      };
-    } catch (error) {
-      this.logger.error(`Could not decrypt message: ${error.message}`);
-      const isOutdatedMessage = error instanceof ProteusErrors.DecryptError.OutdatedMessage;
-      const isDuplicateMessage = error instanceof ProteusErrors.DecryptError.DuplicateMessage;
-
-      if (isOutdatedMessage || isDuplicateMessage) {
-        return {
-          error,
-          isSuccess: false,
-        };
-      }
-
-      throw error;
-    }
+    return this.cryptobox.decrypt(sessionId, messageBytes.buffer);
```

Comment:
Note: Behavior is now changed here, since we did `await ... return` before. Now we just do `return`.